core/script.runtime/src/kotlin/script/dependencies/resolvers.kt

```diff
@@ -32,6 +50,7 @@ interface DependenciesResolver : @Suppress("DEPRECATION") ScriptDependenciesReso
 
     sealed class ResolveResult {
```

Comment:
You're right, we shouldn't mark everything as unresolved in this case. But I still think we should avoid hierarchies here. For example, we can make `isSuccess` a boolean property
```
data class ResolutionResult(
    val isSuccess: Boolean,
    val dependencies: ScriptDependencies,
    val diagnostics: List<ScriptDiagnostic>
)
```